lib/fluent/plugin/kubernetes_metadata_cache_strategy.rb

```diff
@@ -78,13 +79,14 @@ def get_pod_metadata(key, namespace_name, pod_name, record_create_time)
             else
               metadata = {}
             end
+            batch_miss_cache["#{namespace_name}_#{pod_name}"] = metadata
```

Comment:
This appears to be the only place where the batch_miss_cache is populated.  The logic branch on which this resides seems to be for the case where the namespace API called failed and the pod API call succeeded OR failed.

So if we have many logs which have a namespace name that does not exist, we'll have a "batch miss cache" entry for a given namespace/pod pair.

However, if another log entry has the same namespace name, but a different pod name, then 2 API calls will still be made, one for the new pod name and one for the same namespace.

If we remove the batch_miss_cache and instead cache the API calls based on the arguments for a short TTL, we'll avoid heavy API calls in all cases.